include/xrpl/protocol/jss.h

```diff
@@ -747,6 +698,24 @@ JSS(NegativeUNL);                           // out: ValidatorList; ledger type
 #undef TRANSACTION
 #pragma pop_macro("TRANSACTION")
 
+#pragma push_macro("LEDGER_ENTRY")
+#pragma push_macro("LEDGER_ENTRY_DUPLICATE")
+#undef LEDGER_ENTRY
+#undef LEDGER_ENTRY_DUPLICATE
+
+#define LEDGER_ENTRY(tag, value, name, rpcName, fields) \
+    JSS(name);                                          \
+    JSS(rpcName);
+
+#define LEDGER_ENTRY_DUPLICATE(tag, value, name, rpcName, fields) JSS(rpcName);
+
+#include <xrpl/protocol/detail/ledger_entries.macro>
+
+#undef LEDGER_ENTRY
+#undef LEDGER_ENTRY_DUPLICATE
+#pragma pop_macro("LEDGER_ENTRY")
+#pragma pop_macro("LEDGER_ENTRY_DUPLICATE")
```

Comment:
The pattern we're going with is this:

```
#pragma push_macro
#undef
#pragma push_macro
#undef

#define
#define

#include

#undef
#pragma pop_macro
#undef
#pragma pop_macro
```

@Bronek feels more strongly about this pattern, but I agree it should be consistent everywhere.